File: src/main/java/net/ndrei/teslacorelib/gui/IGuiContainerPiece.java
package net.ndrei.teslacorelib.gui;

/**
 * Created by CF on 2016-12-21.
 */
public interface IGuiContainerPiece {
    int getLeft();

    int getTop();

    int getWidth();

    int getHeight();

    void mouseClicked(BasicTeslaGuiContainer container, int mouseX, int mouseY, int mouseButton);

//    void mouseClickMove(BasicTeslaGuiContainer container, int mouseX, int mouseY, int clickedMouseButton, long timeSinceLastClick);

//    void mouseReleased(BasicTeslaGuiContainer container, int mouseX, int mouseY, int state);

    void drawBackgroundLayer(BasicTeslaGuiContainer container, int guiX, int guiY, float partialTicks, int mouseX, int mouseY);

    void drawMiddleLayer(BasicTeslaGuiContainer container, int guiX, int guiY, float partialTicks, int mouseX, int mouseY);

    void drawForegroundLayer(BasicTeslaGuiContainer container, int guiX, int guiY, int mouseX, int mouseY);

    void drawForegroundTopLayer(BasicTeslaGuiContainer container, int guiX, int guiY, int mouseX, int mouseY);

    boolean isVisible();
    void setVisibility(boolean isVisible);
}

File: src/main/java/net/ndrei/teslacorelib/gui/IGuiContainerPiece.kt
package net.ndrei.teslacorelib.gui

/**
 * Created by CF on 2017-06-28.
 */
interface IGuiContainerPiece {
    val left: Int
    val top: Int
    val width: Int
    val height: Int

    fun mouseClicked(container: BasicTeslaGuiContainer<*>, mouseX: Int, mouseY: Int, mouseButton: Int)

    fun drawBackgroundLayer(container: BasicTeslaGuiContainer<*>, guiX: Int, guiY: Int, partialTicks: Float, mouseX: Int, mouseY: Int)

    fun drawMiddleLayer(container: BasicTeslaGuiContainer<*>, guiX: Int, guiY: Int, partialTicks: Float, mouseX: Int, mouseY: Int)

    fun drawForegroundLayer(container: BasicTeslaGuiContainer<*>, guiX: Int, guiY: Int, mouseX: Int, mouseY: Int)

    fun drawForegroundTopLayer(container: BasicTeslaGuiContainer<*>, guiX: Int, guiY: Int, mouseX: Int, mouseY: Int)

    val isVisible: Boolean
    fun setVisibility(isVisible: Boolean)
}
